appinterface/projection/crossfire/crossfire.go

```diff
@@ -538,6 +560,59 @@ func (projection *Crossfire) updateTxSentCount(
 	return nil
 }
 
+func (projection *Crossfire) computeCommitmentRank(
+	rankKey string,
+	crossfireValidatorStatsPrefix string,
+	crossfireValidatorStatsView *view.CrossfireValidatorsStats,
+	crossfireValidatorView *view.CrossfireValidators,
+) error {
+	participants, err := projection.Client.Participants()
+	if err != nil {
+		return fmt.Errorf("error getting participants list %v", err)
+	}
+
+	participantsMap := make(map[string]bool)
+	for _, p := range *participants {
+		participantsMap[p.OperatorAddress] = true
+	}
+
+	validatorCommits, err := crossfireValidatorStatsView.FindAllLike(crossfireValidatorStatsPrefix)
+	if err != nil {
+		return fmt.Errorf("error getting validators' %s number from stats %v", crossfireValidatorStatsPrefix, err)
+	}
+
+	var participatedValidatorCommits []view.CrossfireValidatorsStatsRow
+	for _, v := range validatorCommits {
+		statsOperatorAddress := strings.Split(v.Key, constants.DB_KEY_SEPARATOR)[1]
+		if participantsMap[statsOperatorAddress] {
+			participatedValidatorCommits = append(participatedValidatorCommits, v)
+		}
+	}
+
+	sort.SliceStable(participatedValidatorCommits, func(i, j int) bool {
+		return participatedValidatorCommits[i].Value > participatedValidatorCommits[j].Value
+	})
+
+	rank := 1
+	for index, v := range participatedValidatorCommits {
```

Comment:
Explanation: After sorted and filtered, a single loop could do the dense ranking.